servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/ConnectionClosedAfterResponseTest.java

```diff
@@ -0,0 +1,268 @@
+import static io.servicetalk.transport.netty.internal.AddressUtils.localAddress;
+import static io.servicetalk.transport.netty.internal.BuilderUtils.serverChannel;
+import static io.servicetalk.transport.netty.internal.EventLoopAwareNettyIoExecutors.toEventLoopAwareNettyIoExecutor;
+import static io.servicetalk.transport.netty.internal.GlobalExecutionContext.globalExecutionContext;
+import static java.lang.Integer.MAX_VALUE;
+import static java.nio.charset.StandardCharsets.US_ASCII;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThrows;
+
+public class ConnectionClosedAfterResponseTest {
+
+    @Rule
+    public final Timeout timeout = new ServiceTalkTestTimeout();
+
+    private final ServerSocketChannel server;
+    private final BlockingHttpClient client;
+    private final AtomicReference<CharSequence> encodedResponse = new AtomicReference<>();
+    private final CountDownLatch connectionClosedLatch = new CountDownLatch(1);
+
+    public ConnectionClosedAfterResponseTest() {
+        EventLoopAwareNettyIoExecutor eventLoopAwareNettyIoExecutor =
+                toEventLoopAwareNettyIoExecutor(globalExecutionContext().ioExecutor());
+        EventLoop loop = eventLoopAwareNettyIoExecutor.eventLoopGroup().next();
+
+        ServerBootstrap bs = new ServerBootstrap();
+        bs.group(loop);
+        bs.channel(serverChannel(loop, InetSocketAddress.class));
+        bs.childHandler(new ChannelInitializer() {
+            @Override
+            protected void initChannel(final Channel ch) {
+                ch.pipeline().addLast(new HttpRequestDecoder());
+                ch.pipeline().addLast(new HttpObjectAggregator(MAX_VALUE));
+                ch.pipeline().addLast(new ChannelInboundHandlerAdapter() {
+                    @Override
+                    public void channelRead(final ChannelHandlerContext ctx, final Object msg) {
+                        if (msg instanceof FullHttpRequest) {
+                            ctx.writeAndFlush(ByteBufUtil.writeAscii(ctx.alloc(), encodedResponse.get()))
+                                    .addListener(ChannelFutureListener.CLOSE);
+                        }
+                        ctx.fireChannelRead(msg);
+                    }
+                });
+            }
+        });
+        bs.childOption(AUTO_READ, true);
+        bs.childOption(ALLOW_HALF_CLOSURE, true);
+        bs.childOption(AUTO_CLOSE, false);
+        server = (ServerSocketChannel) bs.bind(localAddress(0))
+                .syncUninterruptibly().channel();
+
+        client = HttpClients.forSingleAddress(HostAndPort.of(server.localAddress()))
+                .protocols(h1().allowChunkedResponseWithoutBody().build())
+                .buildBlocking();
+    }
+
+    @After
+    public void turnDown() throws Exception {
+        try {
+            client.closeGracefully();
```

Comment:
Is this also tearing down the `eventLoopAwareNettyIoExecutor` ? If not we need to do this as well. That said if that's the case we should close the `server` first as otherwise the `EventLoop` will be shutdown already

Reply:
Both client and server use `GlobalExecutionContext` that never shuts down.
Client does not take ownership of `IoExecutor`, neither for provided one nor for global.